walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -72,12 +53,54 @@
           conflicting_events.add(event);
         }
       }
-      //if there were none, return whole day
+      Collections.sort(conflicting_events);
+      ArrayList<Event> merged = mergeEvents(conflicting_events);
+      if (optional.size() > 0) {
+          int slots = getAvailableTimes(conflicting_events, request).size();
+          for (Enumeration orderedOptional = orderOptionalPeople(events, optional); orderedOptional.hasMoreElements() ;) {
+              ArrayList<String> name = new ArrayList<>();
+              name.add((String)orderedOptional.nextElement());
+              ArrayList<Event> temporary = new ArrayList<>();
+              for (Event event : conflicting_events) {
+                  temporary.add(event);
+              }
+              for (Event event : events) {
+                  ArrayList<String> modifier = new ArrayList<String>(event.getAttendees());
+                  modifier.retainAll(name);
+                  if (modifier.size() >= 1) {
+                    temporary.add(event);
+                  }
+              }
+              merged = mergeEvents(temporary);
+              Collections.sort(merged);
+              if (getAvailableTimes(merged, request).size() != 0) {
+                conflicting_events = merged;
+              }
+          }
+      }
       if (conflicting_events.size() == 0) {
-        return Arrays.asList(TimeRange.WHOLE_DAY);
+          if (optional.size() == 0)
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+          else
+            return Arrays.asList();
+      }
+      return getAvailableTimes(conflicting_events, request);
+  }
+
+  public Enumeration orderOptionalPeople(Collection<Event> events, Collection<String> optional) {
+      Hashtable<String, Integer> dict = new Hashtable<String, Integer>();
+      for(String person : optional) {
+          dict.put(person, 0);
+      }
+      for(Event event : events) {
+          ArrayList<String> modifier = new ArrayList<String>(event.getAttendees());
+          modifier.retainAll(optional); //modifier at this point is a list of the people that are optional that have this event
+          for (String person : modifier) {
+              Integer count = dict.get(person);
+              dict.put(person, count+event.getWhen().duration());
+          }
       }
-      ArrayList<Event> merged_events = mergeEvents(conflicting_events);
-      return getAvailableTimes(merged_events, request); 
+      return dict.keys();
```

Comment:
This method name is `orderOptionalPeople` while there is no order action.
and If it just returns person in modifier, maybe you don't need use `dict`?